src/app/data_harvester.rs

```diff
@@ -288,17 +293,96 @@ impl DataCollector {
 
         self.update_cpu_usage();
         self.update_memory_usage();
-        self.update_processes();
         self.update_temps();
-        self.update_network_usage();
-        self.update_disks();
-
         #[cfg(feature = "battery")]
         self.update_batteries();
+        #[cfg(feature = "gpu")]
+        self.update_gpus(); // update_gpus before procs for gpu_pids but after temp/batteries/cpu_usage for appending
+        self.update_processes();
+        self.update_network_usage();
+        self.update_disks();
 
         // Update times for future reference.
         self.last_collection_time = self.data.collection_time;
     }
+    #[cfg(feature = "gpu")]
+    #[inline]
+    fn update_gpus(&mut self) {
+        if self.widgets_to_harvest.use_gpu {
+            let use_temp = self.widgets_to_harvest.use_temp;
+            let use_mem = self.widgets_to_harvest.use_mem;
+            let use_proc = self.widgets_to_harvest.use_proc;
+            let use_cpu = self.widgets_to_harvest.use_cpu;
+            let use_battery = self.widgets_to_harvest.use_battery;
+            #[cfg(feature = "nvidia")]
+            if let Some(data) = nvidia::get_nvidia_vecs(
+                &self.temperature_type,
+                &self.filters.temp_filter,
+                use_temp,
+                use_mem,
+                use_proc,
+                use_cpu,
+                use_battery,
+            ) {
+                if use_temp {
+                    if let Some(mut temp) = data.temperature {
+                        if let Some(ref mut sensors) = self.data.temperature_sensors {
+                            sensors.append(&mut temp);
+                        } else {
+                            self.data.temperature_sensors = Some(temp);
+                        }
+                    }
+                }
+                if use_mem {
+                    if let Some(mem) = data.memory {
+                        self.data.gpu = Some(mem);
+                    }
+                }
+                if use_proc {
+                    if let Some(proc) = data.procs {
+                        self.gpu_pids = Some(proc);
+                    }
+                }
+                if use_cpu {
+                    if let Some(mut cpu) = data.usage {
+                        if let Some(ref mut cpus) = self.data.cpu {
+                            cpus.append(&mut cpu);
+                        } else {
+                            self.data.cpu = Some(cpu);
+                        }
+                    }
+                }
```

Comment:
Not really a fan of this, again, probably better off putting this as a new widget, rather than combining data sources. Apologies if doing that is really hard/confusing, been looking at making new widgets easier...

Reply:
I am fine with removing it. It can be added in a different widget later...